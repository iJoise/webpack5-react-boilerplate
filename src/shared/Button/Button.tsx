import React, { ButtonHTMLAttributes, DetailedHTMLProps } from "react";
import s from "./style/Button.module.scss";

type DefaultButtonPropsType = DetailedHTMLProps<
  ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
>;

type SuperButtonPropsType = DefaultButtonPropsType & {
  rounded?: boolean;
  width?: string | number | undefined;
};

export const Button: React.FC<SuperButtonPropsType> = React.memo(
  ({ children, className, rounded = false, width, ...restProps }) => {
    const widthBtn = {
      width,
    };

    const finalClassName = `${className} ${s.button} ${
      rounded ? s.button__rounded : s.button__default
    }`;

    return (
      <button type="button" className={finalClassName} style={widthBtn} {...restProps}>
        {children}
      </button>
    );
  },
);
