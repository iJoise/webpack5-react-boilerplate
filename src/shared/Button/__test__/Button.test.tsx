import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { Button } from "../Button";

test("button should be render", () => {
  const { container, rerender } = render(<Button>set color</Button>);

  expect(screen.getByRole("button", { name: /set color/i })).toBeInTheDocument();
  expect(container.firstChild).toHaveClass("button");
  expect(container.firstChild).toHaveClass("button__default");

  rerender(<Button rounded>set color</Button>);

  expect(container.firstChild).toHaveClass("button__rounded");

  expect(container).toMatchSnapshot();
});
