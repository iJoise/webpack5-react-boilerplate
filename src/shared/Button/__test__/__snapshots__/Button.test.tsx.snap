// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should be render 1`] = `
<div>
  <button
    class=" button button__rounded"
    type="button"
  >
    set color
  </button>
</div>
`;
